### PR TITLE
use the code cache directory for the code cache file lock

### PR DESCRIPTION
**Pull request by ahnitz**

### CODE REVIEW

File: pycbc/weave.py [hunk 1]
```diff
@@ -37,14 +38,13 @@ def pycbc_compile_function(code,arg_names,local_dict,global_dict,
     """
     from scipy.weave.inline_tools import _compile_function
     headers = [] if headers is None else headers
-    lockfile_dir = os.path.dirname(module_dir)
-    lockfile_name = os.path.join(lockfile_dir, 'code_lockfile')
-    print("attempting to aquire lock '%s' for compiling code" % lockfile_name)
+    lockfile_name = os.path.join(pycbc._cache_dir_path, 'code_lockfile')
+    logging.info("attempting to aquire lock '%s' for compiling code" % lockfile_name)
```
Comment (ahnitz) [inline pycbc/weave.py]: __[flake8]__
* [E501] line too long (85 > 79 characters)

**Comment by spxiwh** [comment]
I think this needs rebasing.

**Comment by ahnitz** [comment]
Ok, it's been rebased.

File: pycbc/weave.py [hunk 2]
```diff
@@ -37,14 +38,14 @@ def pycbc_compile_function(code,arg_names,local_dict,global_dict,
     """
     from scipy.weave.inline_tools import _compile_function
     headers = [] if headers is None else headers
-    lockfile_dir = os.path.dirname(module_dir)
+    lockfile_dir = pycbc._cache_dir_path
     lockfile_name = os.path.join(lockfile_dir, 'code_lockfile')
-    print("attempting to aquire lock '%s' for compiling code" % lockfile_name)
+    logging.info("attempting to aquire lock '%s' for compiling code" % lockfile_name)
```
Comment (ahnitz) [inline pycbc/weave.py]: __[flake8]__
* [E501] line too long (85 > 79 characters)

Comment (spxiwh) [inline pycbc/weave.py]: Why not fix this?

**Pull request merged**